middleware.go

```diff
@@ -27,6 +27,15 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+// InitEntry initializes the log entry in context
+func InitEntry(ctx context.Context, logger *logrus.Logger) context.Context {
```

Comment:
I feel like this should be called WithLogger.  Having entry in the name when there's no Entry in the function signature is weird to me.